portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/*
+ * Dynamically generate new html content when button is pressed
+ */
+async function getBlogComments() {
+  const response = await fetch('/data');
+  const blogData = await response.json();
+
+  const blogListElement = document.getElementById('blog-container');
+  addAllBlogComments(blogListElement, blogData);
+}
+
+function createListElement(text) {
+  const liElement = document.createElement('li');
+  liElement.innerText = text;
+  return liElement;
+}
+
+function addAllBlogComments(blogListElement, blogData) {
+  const listLength = blogData.commentsList.length;
+  for (let i = 0; i < listLength; i++) {
```

Comment:
This loop can be simplified by using a for...of loop (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)

```
for (const comment of blogData.commentsList) {
  blogListElement.appendChild(createListElement(comment));
}
```